src/Domain/CoordinateMaps/CylindricalEndcap.hpp

```diff
@@ -0,0 +1,150 @@
+ * \f$(\bar{x},\bar{y})\f$ are mapped to portions of lines that pass
+ * through \f$P\f$. Along each of these curves, \f$\bar{z}=-1\f$ is
+ * mapped to a point on sphere 1 and \f$\bar{z}=+1\f$ is mapped to a
+ * point on sphere 2.
+ *
+ * CylindricalEndcap is intended to be composed with Wedge2D maps to
+ * construct a portion of a cylindrical domain for a binary system.
+ *
+ * CylindricalEndcap is described briefly in the Appendix of
+ * Phys. Rev. D 86, 084033 (2012), https://arxiv.org/abs/1206.3015.
+ * CylindricalEndcap is used to construct the blocks labeled 'CA
+ * wedge', 'EA wedge', 'CB wedge', 'EE wedge', and 'EB wedge' in
+ * Figure 20 of that paper.
+ *
+ * CylindricalEndcap is implemented using `FocallyLiftedMap`
+ * and `FocallyLiftedEndcap`; see those classes for details.
+ *
+ * ### Restrictions on map parameters.
+ *
+ * We demand that Sphere 1 is fully contained inside Sphere 2. It is
+ * possible to construct a valid map without this assumption, but the
+ * assumption simplifies the code, and the expected use cases obey
+ * this restriction.
+ *
+ * We also demand that \f$z_\mathrm{P} > C_1^2\f$, that is, the plane
+ * in the above and below figures lies to the right of \f$C_1^2\f$.
+ * This restriction not strictly necessary but is made for simplicity.
+ *
+ * The map is invertible only for some choices of the projection point
+ * \f$P\f$.  Given the above restrictions, the allowed values of
+ * \f$P\f$ are illustrated by the following diagram:
+ *
+ * \image html CylindricalEndcap_Allowed.svg "Allowed region for P." width=75%
+ *
+ * The plane \f$z=z_\mathrm{P}\f$ intersects sphere 1 on a circle. The
+ * cone with apex \f$C_1\f$ that intersects that circle has opening
+ * angle \f$2\theta\f$ as shown in the above figure. Construct another
+ * cone, the "invertibility cone", with apex \f$S\f$ chosen such that
+ * the two cones intersect at right angles on the circle; thus the
+ * opening angle of the invertibility cone is \f$2(\pi-\theta)\f$.  A
+ * necessary condition for invertibility is that the projection point
+ * \f$P\f$ lies inside the invertibility cone, but not between \f$S\f$
+ * and sphere 1.  Placing the projection point \f$P\f$ to the right of
+ * \f$S\f$ (but inside the invertibility cone) is ok for
+ * invertibility.
+ *
+ * In addition to invertibility and the two additional restrictions
+ * already mentioned above, we demand a few more restrictions on the
+ * map parameters to simplify the logic for the expected use cases, and
+ * to ensure that jacobians do not get too large. We demand:
+ *
+ * - \f$P\f$ is not too close to the edge of the invertibility cone.
+ * - \f$P\f$ is contained in sphere 2.
+ * - \f$P\f$ is to the left of \f$z_\mathrm{P}\f$.
+ * - \f$z_\mathrm{P}\f$ is not too close to the center or the edge of sphere 1.
+ * - If a line segment is drawn between \f$P\f$ and any point on the
+ *   intersection circle, the angle between the line segment and
```

Comment:
Is the intersection circle the surface where \bar{z}=-1 gets mapped to? If so, probably good to say that. If not, I got confused by the docs...please clarify where the intersection surface is either way?